File: app/src/main/java/com/github/alexeygorovoy/moxytemplate/dagger/app/AppContextModule.java
package com.github.alexeygorovoy.moxytemplate.dagger.app;

import android.content.Context;

import dagger.Module;
import dagger.Provides;


@Module
public class AppContextModule {


    private final Context context;

    public AppContextModule(Context aContext) {
        this.context = aContext;
    }

    @AppScope
    @Provides
    Context provideAppContext() {
        return context;
    }

}

File: app/src/main/java/com/github/alexeygorovoy/moxytemplate/dagger/app/AppContextModule.kt
package com.github.alexeygorovoy.moxytemplate.dagger.app

import android.content.Context

import dagger.Module
import dagger.Provides

@Module
class AppContextModule(private val context: Context) {

    @AppScope
    @Provides
    internal fun provideAppContext(): Context {
        return context
    }
}
